posts/2024-02-15-waku-network-tech-overview.mdx

```diff
@@ -0,0 +1,178 @@
+- Reliability: gossipsub has built-in redundancy, which gives fair reliability when operating in a decentralized peer-to-peer network, where no node can be trusted to be reliable or well-behaved.
+- Anonymity: as nodes forward messages from other nodes in their mesh, and no metadata is present on individual messages (e.g., no plain text signature), this provides fair anonymity as it is not possible for an observer to know if their neighbor originated or forwarded a message. This works when combined with a decentralized peer discovery mechanism such as discv5.
+
+We named Waku’s usage of gossipsub, the protocol and network. **Waku Relay.**
+
+### Message Routing - Sharding
+
+One of the drawbacks of Gossipsub is that every node in the network receives and sends every message of the network, with some amplification.
+
+One can see how this limits scalability: it cannot be expected to traffic the whole traffic of the network from a household internet connection.
+
+To solve this problem, sharding is used: instead of having one gossipsub network. The Waku network is split into several networks, or shards.
+Currently, the Waku network is split into 8 shards.
+It means that any user of a Waku application would only relay the traffic of one shard, or one 8th (approximately) of the whole network.
+
+8 was a small arbitrary number to start with. From our theoretical analysis, we believe that a shard can support around 10k active users, meaning 80k for the whole network. We are working on running further simulations to confirm our assumptions. We are also onboarding developers and users on the network as a way to test the theory.
+The aim is to increase the number of shards in the network over time.
+
+Thanks to the enhancements made to discv5, nodes are able to know what shards other peers are servicing before connecting to them.
+
+### Message Routing - Autosharding
+
+One issue with sharding is that users and applications need to know what shard to use.
+
+One application could arbitrarily decide what shard they want to use but that may not scale easily, especially when adding new shards. It is also one more decision for the developer to make; we prefer to make the dev ex as easy as possible. It also gives the opportunity for developers to build applications that spread on several shards without delegating shard choices to the user.
+
+Autosharding is a simple protocol that dispatches messages on a shard based on the application.
+
+### Message Routing - RLN Relay
+
+Waku is generalized, meaning any kind of payload can be transported. Hence, there is no strict definition of “spam”. Whether the message contains a [meme](https://github.com/b4s36t4/waku-meme) or a [zk note for private DeFi](https://railgun.ch/#/), Waku shouldn't even know the content; metadata and data should remain private.
```

Comment:
"Waku is generalized _and privacy-preserving_"
We do define spam (as high rate messages). There is no strict definition of "spam content".